src/Tasks/WriteCodeFragment.cs

```diff
@@ -245,5 +274,312 @@ private string GenerateCode(out string extension)
+        /// Returns an array of types with a length equal to the number of positional parameters.
+        /// If no suitable constructor is found, the array will contain null types.
+        /// </summary>
+        private Type[] FindPositionalParameterTypes(Type attributeType, IReadOnlyList<AttributeParameter> positionalParameters)
+        {
+            // The attribute type might not be known.
+            if (attributeType is not null)
+            {
+                List<Type[]> candidates;
+
+                // Find the constructors with the same number
+                // of parameters as we will be specifying.
+                candidates = attributeType
+                    .GetConstructors()
+                    .Select(c => c.GetParameters().Select(p => p.ParameterType).ToArray())
+                    .Where(t => t.Length == positionalParameters.Count)
+                    .ToList();
+
+                if (candidates.Count == 1)
+                {
+                    return candidates[0];
+                }
+                else if (candidates.Count > 1)
+                {
+                    Log.LogMessageFromResources("WriteCodeFragment.MultipleConstructorsFound");
+
+                    // Before parameter types could be specified, all parameter values were
+                    // treated as strings. To be backward-compatible, we need prefer the
+                    // constructor that has all string parameters, if it exists.
+                    var allStringParameters = candidates.FirstOrDefault(c => c.All(t => t == typeof(string)));
+
+                    if (allStringParameters is not null)
+                    {
+                        return allStringParameters;
+                    }
+
+                    // There isn't a constructor where all parameters are strings, so we are free
+                    // to pick one of the candidates. To ensure that we always select the same
+                    // constructor, sort the candidates by type names, then pick the first one.
+                    return candidates
+                        .OrderBy(c => string.Join(",", c.Select(t => t.FullName)))
+                        .First();
+                }
+            }
+
+            // If a matching constructor was not found, or we don't
+            // know the attribute type, then return an array of null
+            // types to indicate that each parameter type is unknown.
+            return positionalParameters.Select(x => default(Type)).ToArray();
+        }
+
+        /// <summary>
+        /// Attempts to convert the raw value provided in the metadata to the type with the specified name.
+        /// Returns true if conversion is successful. An error is logged and false is returned if the conversion fails.
+        /// </summary>
+        private bool TryConvertParameterValue(string typeName, string rawValue, out CodeExpression value)
+        {
+            var parameterType = Type.GetType(typeName);
```

Comment:
Can this throw?

Reply:
`Type.GetType(string)` will return null if the type is not found.